src/main.js

```diff
@@ -41,42 +45,59 @@ document.addEventListener('DOMContentLoaded', () => {
   handleLocation();
 });
 
-export const getUrl = (event) => {
-  event.preventDefault();
-  const url = new URL(event.currentTarget.href);
+export const getUrl = (stringUrl) => {
+  const url = new URL(stringUrl);
+
   url.pathname = `${basePath}${url.pathname}`;
   return url;
 };
 
-export const setParams = (event, url) => {
-  const dataset = event.currentTarget.dataset;
+export const setParams = (queries = {}, url) => {
   const searchKeys = Object.values(SEARCH_PARAMS);
-  Object.keys(dataset).forEach((key) => {
+  Object.keys(queries).forEach((key) => {
     if (searchKeys.includes(key)) {
-      url.searchParams.set(key, dataset[key]);
+      url.searchParams.set(key, queries[key]);
     }
   });
 };
 
 export const handleLocation = async () => {
-  const { pathname, search } = window.location;
+  const { search } = window.location;
```

Comment:
this change brokes the logic, be carefull when you change functions used all over the project

Reply:
Just pathname is not used in this function